mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstract.php

```diff
@@ -2105,19 +2105,40 @@ protected function get_recurring_total() {
 	}
 
 	/**
-	 * Get available payment methods in checkout based on amount, currency and sequenceType
+	 * Check if payment method is available in checkout based on amount, currency and sequenceType
 	 *
 	 * @param $filters
 	 *
 	 * @return bool
 	 */
-	protected function getAvailableMethodsInCheckout( $filters ) {
+	protected function isAvailableMethodInCheckout( $filters ) {
 
 		$settings_helper = Mollie_WC_Plugin::getSettingsHelper();
 		$test_mode       = $settings_helper->isTestModeEnabled();
 
 		try {
-			$methods = Mollie_WC_Plugin::getApiHelper()->getApiClient( $test_mode )->methods->all( $filters );
+
+			$filtersKey   = $filters['amount']['currency'] . '_' . str_replace( '.', '', $filters['amount']['value'] ) . '_' . $filters['sequenceType'];
+			$transient_id = Mollie_WC_Plugin::getDataHelper()->getTransientId( 'api_methods_' . ( $test_mode ? 'test' : 'live' ) . '_' . $filtersKey );
```

Comment:
Nice, this will work! It will result in a lot of caches. Do you know whether these are automatically cleaned up?

Reply:
Yes, I've considered this. WordPress will not automatically remove expired transients so:
- if the same cache is found but invalid, I'll delete it (delete_transcient). 
- on database updates WordPress will also delete all expired transients
https://core.trac.wordpress.org/changeset/25838